File: app/channels/ticket_channel.rb
class TicketChannel < ApplicationCable::Channel
  def subscribed
    ticket = Ticket.find(params[:id])
    ap "je suis dans #{__method__}"
    ap ticket
    stream_for ticket
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end
end

File: app/javascript/channels/ticket_channel.js
import consumer from "./consumer"




document.addEventListener('turbolinks:load', () => {
  if (document.querySelector('#ticket-show')) {
    // console.log("je suis dans mon izefzefezf")
    const ticketId = document.querySelector('#ticket-show').dataset.ticketId

    const ticketChannel = consumer.subscriptions.create({ channel: "TicketChannel", id: ticketId }, {
      received(data) {
        // data => { sent_by: "Paul", body: "This is a cool ticket app." }
        // console.log("je suis dans receive data")
        // console.log(data)
        if (data.action == 'refresh') {
          document.location.reload()
        }
      }
    })

    // ticketChannel.send({ sent_by: "Paul", body: "This is a cool ticket app." })
  }
});




